client/my-sites/plan-features/index.jsx

```diff
@@ -753,14 +754,18 @@ export class PlanFeatures extends Component {
 	}
 
 	renderPlanFeatureColumns( rowIndex ) {
-		const { planProperties, selectedFeature, withScroll } = this.props;
+		const { planProperties, selectedFeature, withScroll, hideCustomDomainFeature } = this.props;
 
 		return map( planProperties, ( properties ) => {
 			const { features, planName } = properties;
 
 			const featureKeys = Object.keys( features );
 			const key = featureKeys[ rowIndex ];
-			const currentFeature = features[ key ];
+			let currentFeature = features[ key ];
+
+			if ( hideCustomDomainFeature && FEATURE_CUSTOM_DOMAIN === currentFeature?.getSlug() ) {
+				currentFeature = null;
+			}
```

Comment:
I wonder if this might fit better under `renderFeatureItem` which I think should resolve the issue (though I've not tested it yet).

https://github.com/Automattic/wp-calypso/blob/b703449c2b079916da70356acd66ca823760cebf/client/my-sites/plan-features/index.jsx#L730-L753

Currently, the change is not being checked in `renderMobileFeatures` which is causing the issue.

https://github.com/Automattic/wp-calypso/blob/b703449c2b079916da70356acd66ca823760cebf/client/my-sites/plan-features/index.jsx#L405-L409

Reply:
Thanks for reviewing, and good catch with this not working on mobile! This prompted me to spend more time on this, and I think we can safely modify the `features` array.